mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -91,7 +94,13 @@
                 attachments.addAll(listAttachments((Multipart) entity.getBody(), Context.fromEntity(entity)));
             } else {
                 if (isAttachment(entity, context)) {
-                    attachments.add(retrieveAttachment(messageWriter, entity));
+                    try {
+                        attachments.add(retrieveAttachment(messageWriter, entity));
```

Comment:
I think _retrieveAttachment_ should not throw when:
- inline attachment without cid
- unknown content-disposition